### ISSUE TITLE
Tokenization around "-" and "_"?

### ISSUE THREAD

**Issue by rluba** [typesense/typesense#122]
## Description

One of the fields of my collection is configured as a "string" and contains filenames. But when searching for documents, partial hits within the filenames are not returned at all. It seems like Typesense does not break apart tokens at underscores or dashes.

For example, a file named "something_2020-07-13_with_important_words.pdf" _cannot be found_ by searching for "important", "words" or "2020-07-13". It _can_ be found by searching for "something", but that’s about it.

Is my assumption about tokenization correct? Is there a way to influence tokenization?

## Metadata

**Typsense Version**: 0.14.0

**OS**: Ubuntu 20.04 LTS

**Comment by kishorenc** [comment]
Yes, Typesense currently just removes all non-alphanumeric characters in the ASCII range from the input string. Definitely need to create a list of special tokens that can be specified to be treated as "spaces" instead (at a field level), thereby forcing a tokenization on them.

Similar to https://github.com/typesense/typesense/issues/95

**Comment by kishorenc** [comment]
As a workaround you have to add the spaces before indexing.

**Comment by kishorenc** [comment]
Thank you @baltpeter, this is great, detailed feedback. 

I've fixed this issue in `typesense/typesense:0.22.0.rcu8` and also added some tests with those examples. I will appreciate another look. Thanks!

**Comment by baltpeter** [comment]
Awesome, I can confirm that the problem I described has been fixed in `typesense/typesense:0.22.0.rcu8`. Thanks, @kishorenc! 

I also did some more testing and couldn't find any more edge cases.  
The only thing left that's slightly unfortunate, is that `fluxfail` now won't match `flux.fail` anymore (with `token_separators: ['.']`), whereas it previously did as it recognized the missing `.` as a typo. But that's probably covered by #133, isn't it?
